include/swift/SIL/SILType.h

```diff
@@ -553,33 +547,346 @@ inline llvm::raw_ostream &operator<<(llvm::raw_ostream &OS, SILType T) {
   return OS;
 }
 
-inline SILType SILParameterInfo::getSILType() const {
-  if (isIndirect()) {
-    return SILType::getPrimitiveAddressType(getType());
-  } else {
-    return SILType::getPrimitiveObjectType(getType());
+/// Transient wrapper for SILParameterInfo and SILResultInfo conventions. This
+/// abstraction helps handle the transition from canonical SIL conventions to
+/// lowered SIL conventions.
+class SILModuleConventions {
+  friend SILParameterInfo;
+  friend SILResultInfo;
+  friend SILFunctionConventions;
+
+  static bool isIndirectSILParam(SILParameterInfo param, bool loweredAddresses);
```

Comment:
Why do we need these static methods? When will we not have access to SILModuleConventions and not have access to lowered addresses?